network/default.go

```diff
@@ -1045,9 +1046,51 @@ func (n *network) processNetChan(listener tunnel.Listener) {
 
 				// add all the routes we have received in the sync message
 				for _, pbRoute := range pbNetSync.Routes {
+					// unmarshal the routes received from remote peer
 					route := pbUtil.ProtoToRoute(pbRoute)
-					if err := n.router.Table().Create(route); err != nil && err != router.ErrDuplicateRoute {
-						log.Debugf("Network node %s failed to add route: %v", n.id, err)
+					// lookup the best route for the given service in our routing table
+					q := []router.QueryOption{
+						router.QueryService(route.Service),
+						router.QueryStrategy(router.AdvertiseBest),
```

Comment:
Use advertising strategy already set in the router

Reply:
Fixing this now.